File: spring-data-redis-example/src/main/java/com/sunil/redis/ds/maxSumSubArr.java
package com.sunil.redis.ds;

public class maxSumSubArr {

    public static void main(String[] args) {
        int[] nums = {2, 3, -4, 1, 2, -1, -5};
        System.out.println(maxSum(nums));
    }
    public static int maxSum(int[] nums) {
        int maxSum = 0;
        int curSum = 0;

        for (int num : nums) {
            if (curSum < 0)
                curSum = 0;
            curSum = curSum + num;
            maxSum = Math.max(curSum, maxSum);
        }
        return maxSum;
    }
}

File: python_75/two/max_sum_subarr.py
# input = [1, 3, -2, 5, 2, -4]
# output = 9

nums = [1, 3, -2, 5, 2, -4]

curSum = 0
maxSum = 0

for num in nums:
  if curSum < 0:
    curSum = 0 
  curSum = curSum + num
  maxSum = max(maxSum, curSum)

print(maxSum)